homeassistant/components/simplefin/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from simplefin4py import Account
+from simplefin4py.model import AccountType
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import SimpleFinDataUpdateCoordinator
+
+
+def _enum_to_icon(inferred_type: AccountType) -> str:
+    """Based on the inferred account type - guess a starting icon."""
+    if inferred_type == AccountType.CHECKING:
+        return "mdi:checkbook"
+    if inferred_type == AccountType.CREDIT_CARD:
+        return "mdi:credit-card"
+    if inferred_type == AccountType.SAVINGS:
+        return "mdi:piggy-bank-outline"
+    if inferred_type == AccountType.INVESTMENT:
+        return "mdi:chart-areaspline"
+    if inferred_type == AccountType.MORTGAGE:
+        return "mdi:home-city-outline"
+
+    return "mdi:cash"
```

Comment:
I think I can probalby drop this entirely because (due to bad programming) 

`AccountType.MORTGAGE` actually is a string based enum of value: `mdi:home-city-outline`

I got dinged once for doing that in a backing lib ... but it does work.

Reply:
So I could not make this work - I think there _may_ be an error or at least _user error_

I did open a ticket however
https://github.com/home-assistant/core/issues/110127

Reply:
Issue #110127 talks about *attribute* icon translations not working, that's to influence the icon used when displaying entity state *attributes*. That's not relevant here though.